sdks/go/pkg/beam/core/runtime/pipelinex/replace.go

```diff
@@ -243,3 +245,33 @@ func findFreeName(seen map[string]bool, name string) string {
 		}
 	}
 }
+
+// ApplySdkImageOverrides takes a pipeline and a map of patterns to overrides,
+// and proceeds to replace matching ContainerImages in any Environments
+// present in the pipeline.
+func ApplySdkImageOverrides(p *pipepb.Pipeline, patterns map[string]string) error {
```

Comment:
Consider documenting the expectation that any given environment is expected to match only a single pattern for replacement, and that if multiple patterns would match, it's arbitrary which will be applied (due to map iteration ordering being random.)

There's no good way to handle such conflict cases for multple matches. I suspect the most we can do is say it's undefined in the flag itself, as well as the commentary change in the previous paragraph.

Reply:
Done. Also adding this to the flag description.